tasks/libs/ciproviders/github_api.py

```diff
@@ -794,3 +780,25 @@ def ask_review_actor(pr):
     for event in pr.get_issue_events():
         if event.event == "labeled" and event.label.name == "ask-review":
             return event.actor.name or event.actor.login
+
+
+def generate_local_github_token(ctx):
+    """
+    Generates a github token locally.
+    """
+
+    # Try without login first, then with login
+    for login in False, True:
```

Comment:
We don't gain much by doing this and readability takes a hit. Let's instead put the logic for the non-login generation inside the `try` and the login approach during exception handling.